Source/WebCore/animation/KeyframeEffect.cpp

```diff
@@ -2551,6 +2551,11 @@ void KeyframeEffect::computeHasSizeDependentTransform()
 {
     m_animatesSizeAndSizeDependentTransform = (m_blendingKeyframes.hasWidthDependentTransform() && m_blendingKeyframes.containsProperty(CSSPropertyWidth))
         || (m_blendingKeyframes.hasHeightDependentTransform() && m_blendingKeyframes.containsProperty(CSSPropertyHeight));
+
+    if (auto target = targetStyleable()) {
+        if (target->pseudoElementIdentifier && target->pseudoElementIdentifier->pseudoId == PseudoId::ViewTransitionGroup)
+            m_animatesSizeAndSizeDependentTransform |= ((m_blendingKeyframes.containsProperty(CSSPropertyWidth) || m_blendingKeyframes.containsProperty(CSSPropertyHeight)) && m_blendingKeyframes.containsProperty(CSSPropertyTransform));
```

Comment:
Checking just `CSSPropertyTransform` will not capture the individual transform properties `scale`, `translate` and `rotate`. You can use the static function `propertiesContainTransformRelatedProperty()` to check for all transform-related CSS properties.

Reply:
I *think* that's what I want. The default UA generated styles will run an animation of `transform` that animates the element-to-screen transform between the new and old element state. The common case is that this is just animating a translation. The same default generated stylesheet runs an animation of width and height to show any size changes.

If the page has written their own styles targeting `::view-transition-group(name)` which `unset` the animation on `transform` and instead add a new one for one of the individual properties, then they're well outside of the default path. It's less obvious that we want to treat this case any different to any other pairing of transform/size properties in an animation.

Happy to be convinced otherwise though :)

Reply:
Sorry, I hadn't realized this had something to do with some default styles! In that case, I think it would be nice to have a comment to that effect, just to make it clear.

Reply:
+1 to add a comment for clarification :)